core/src/main/java/org/fao/geonet/kernel/security/keycloak/KeycloakHttpSessionManager.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Changes were taken from org.keycloak/keycloak-adapter-core/11.0.1 as there were bugs in current version
+// that we prevending the Admin URL client backchannel logout from working.
+// If upgrading maven dependency then it may be possible that the most recent KeycloakHttpSessionManager can be used
```

Comment:
I updated the comments to make it more clear as to why this was required.
It is mostly because the WebApplicationContextUtils.getWebApplicationContext(servletContext) returns null and it was causing NPE.  It seems like the the application is not registered as a WebApplicationContext so I had to modify the code to use ApplicationContextHolder.get() and this fixed the issue.

Maybe in the future if the application is registered correctly as a WebApplicationContext then this file can be removed.

Is this correction OK - or should there be a fix to register the application correctly as a WebApplicationContext?